resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/Client.scala

```diff
@@ -73,6 +76,30 @@ private[spark] class Client(
   private val driverJavaOptions = sparkConf.get(
     org.apache.spark.internal.config.DRIVER_JAVA_OPTIONS)
 
+  // create resource of kind - SparkJob representing the deployed spark app
+  private val statusMap = Map(
+    STATUS_CREATION_TIMESTAMP -> Calendar.getInstance().getTime().toString(),
+    STATUS_COMPLETION_TIMESTAMP -> STATUS_NOT_AVAILABLE,
+    STATUS_DRIVER -> kubernetesDriverPodName,
+    STATUS_DRIVER_IMAGE -> driverDockerImage,
+    STATUS_EXECUTOR_IMAGE -> sparkConf.get(EXECUTOR_DOCKER_IMAGE),
+    STATUS_JOB_STATE -> JobState.QUEUED,
+    STATUS_DESIRED_EXECUTORS -> sparkConf.getInt("spark.executor.instances", 1),
+    STATUS_CURRENT_EXECUTORS -> 0,
+    STATUS_DRIVER_UI -> STATUS_PENDING
+  )
+
+  // Failure might be due to TPR inexistence or maybe we're stuck in the 10 minute lag
+  // TODO: in the latter case we can attempt a retry depending on the rc
+  // This also assumes that once we fail at creation, we won't bother trying
+  // anything on the resource for the lifetime of the app
+  Try(sparkJobResourceController.createJobObject(kubernetesAppId, statusMap)) match {
+    case Success(_) => sparkConf.set("spark.kubernetes.jobResourceSet", "true")
+      sparkConf.set("spark.kubernetes.jobResourceName", kubernetesAppId)
+    case Failure(_) =>
+      sparkConf.set("spark.kubernetes.jobResourceSet", "false")
```

Comment:
I also don't think the current naming reflects what this does.  Right now it describes the state it expects (the job resource has been set on the cluster) rather than what actions the code takes when the flag is set (create a TPR and report status to it as the job progresses).  Maybe name something like `spark.kubernetes.statusReporting.enabled=true` and `spark.kubernetes.statusReporting.resourceName=<asdf>` ?